tests/test_translating_parser.py

```diff
@@ -0,0 +1,183 @@
+from os.path import join
+
+from prance import _TranslatingParser
+
+
+class SpecificationTester:
+    """
+    Helper class that tests correct reference translation. Loads a specification file from tests/specs, parses it with
+    the TranslatingParser and provides assertion methods:
+    * assert_schemas to check what’s in /components/schemas.
+    * assert_path_ref to check the $ref of the only operation response in the specification.
+    * assert_schema_ref to check the $ref of a schema in /components/schemas.
+
+    Typical usage:
+    tester = SpecificationTester("my_spec_file")
+    tester.assert_path_ref("some_other_refd_file.spec.yaml_SomeObject")
+    tester.assert_schemas(
+        {"some_other_refd_file.spec.yaml_SomeObject", "some_other_refd_file.spec.yaml_SomeOtherObject"}
+    )
+    tester.assert_schema_ref(
+        "some_other_refd_file.spec.yaml_SomeObject", "some_other_refd_file.spec.yaml_SomeOtherObject",
+    )
+    """
+    @staticmethod
+    def _parse_spec(name):
+        file = f"{name}.spec.yaml"
+        path = join("tests", "specs", "translating_parser", file)
+        parser = _TranslatingParser(path)
+        parser.parse()
+        return parser.specification
+
+    @staticmethod
+    def _assert_ref(schema, ref):
+        assert "$ref" in schema
+        assert schema["$ref"] == f"#/components/schemas/{ref}"
+
+    def __init__(self, file):
```

Comment:
this should take specification as parameter

Reply:
That’s pretty bold claim that it should. It’s an option for sure and it can be implemented that way too. However the class is currently used only in a way where the file name part is enough.

Reply:
thats where named ctors come in

Reply:
basically if there is a named ctor that goes from file to specification, then the normal ctor can just take the specification

Reply:
The BaseParser constructor is also not named. It uses two parameters, _url_ and _spec_string_. That may not be a very good practice, but it feels a bit strange to use it here differently.